crypto/rsa/rsa_gen.c

```diff
@@ -186,10 +188,19 @@ static int rsa_builtin_keygen(RSA *rsa, int bits, int primes, BIGNUM *e_value,
             }
             if (!BN_sub(r2, prime, BN_value_one()))
                 goto err;
-            if (!BN_gcd(r1, r2, rsa->e, ctx))
-                goto err;
-            if (BN_is_one(r1))
+            if (BN_mod_inverse(r1, r2, rsa->e, ctx)) {
+               /* GCD == 1 since inverse exists */
                 break;
+            } else {
+                error = ERR_peek_last_error();
+                if (ERR_GET_LIB(error) == ERR_LIB_BN &&
```

Comment:
Style nit.  Put the && at the start of the line, and indent that second line an extra tabstop.